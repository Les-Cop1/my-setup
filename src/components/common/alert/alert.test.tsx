import React from 'react'

import { Alert } from '@components'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'

describe('Alert', () => {
  test('Alert should be defined', () => {
    render(<Alert testid="testId" message="message" />)
    expect(screen.getByTestId('testId')).toBeInTheDocument()
  })

  test('Alert should have correct type color', () => {
    render(<Alert testid="default" message="default" />)
    render(<Alert testid="error" message="error" type="error" />)
    render(<Alert testid="success" message="success" type="success" />)
    render(<Alert testid="info" message="info" type="info" />)
    expect(screen.getByTestId('default')).toHaveClass('bg-red-50')
    expect(screen.getByText('default')).toHaveClass('text-red-800')
    expect(screen.getByTestId('error')).toHaveClass('bg-red-50')
    expect(screen.getByText('error')).toHaveClass('text-red-800')
    expect(screen.getByTestId('success')).toHaveClass('bg-teal-50')
    expect(screen.getByText('success')).toHaveClass('text-teal-800')
    expect(screen.getByTestId('info')).toHaveClass('bg-cyan-50')
    expect(screen.getByText('info')).toHaveClass('text-cyan-800')
  })

  test('Alert should have className', () => {
    render(<Alert testid="testId" message="message" className="test-class" />)
    expect(screen.getByTestId('testId')).toHaveClass('test-class')
  })

  test('Alert should contain text', () => {
    render(<Alert testid="testId" message="hello world" className="test-class" />)
    expect(screen.getByText('hello world')).toBeInTheDocument()
  })

  test('Snapshot', () => {
    const { container } = render(<Alert testid="testId" message="message" />)
    expect(container).toMatchSnapshot()
  })
})
