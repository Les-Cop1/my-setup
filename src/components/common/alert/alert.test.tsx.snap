// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Snapshot 1`] = `
<div>
  <div
    class="rounded-md p-4 bg-red-50"
    testid="testId"
  >
    <div
      class="flex"
    >
      <div
        class="flex-shrink-0"
      >
        <svg
          aria-hidden="true"
          class="h-5 w-5 text-red-800"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-8-5a.75.75 0 01.75.75v4.5a.75.75 0 01-1.5 0v-4.5A.75.75 0 0110 5zm0 10a1 1 0 100-2 1 1 0 000 2z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="ml-3"
      >
        <p
          class="text-sm font-medium text-red-800"
        >
          message
        </p>
      </div>
      <div
        class="ml-auto pl-3"
      >
        <div
          class="-mx-1.5 -my-1.5"
        >
          <button
            class="inline-flex rounded-md p-1.5 focus:outline-none focus:ring-2 focus:ring-offset-2 bg-red-50 text-red-800 hover:bg-red-100 focus:ring-red-600  focus:ring-offset-red-50"
            type="button"
          >
            <span
              class="sr-only"
            >
              Dismiss
            </span>
            <svg
              aria-hidden="true"
              class="h-5 w-5"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.28 5.22a.75.75 0 00-1.06 1.06L8.94 10l-3.72 3.72a.75.75 0 101.06 1.06L10 11.06l3.72 3.72a.75.75 0 101.06-1.06L11.06 10l3.72-3.72a.75.75 0 00-1.06-1.06L10 8.94 6.28 5.22z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
